File: 백준/Silver/25325. 학생 인기도 측정/학생 인기도 측정.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val studentList = readLine().split(' ')
    val studentCount = mutableMapOf<String, Int>()

    for (i in 0 until n) {
        studentCount[studentList[i]] = 0
    }

    for (i in 0 until n) {
        val input = readLine().split(' ')
        for (s in input) {
            studentCount[s] = studentCount[s]!! + 1
        }
    }

    val sortedStudentCount = studentCount.entries.sortedByDescending { it.value }

    println(sortedStudentCount.joinToString("\n") { "${it.key} ${it.value}" })
}

File: 백준/Silver/25325. 학생 인기도 측정/학생 인기도 측정.js
const solution = (arr) => {
  const studentCount = {};

  list.forEach((x) => {
    studentCount[x] = 0;
  })

  arr.forEach((x) => {
    studentCount[x]++;
  });

  const sortStudentCount = Object.entries(studentCount).sort((x, y) => y[1] - x[1]);

  return sortStudentCount.map(([student, count]) => `${student} ${count}`).join('\n');
};

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let N = 0;
let list = [];
const arr = [];

rl.on("line", function (line) {
  if (N === 0) N = +line;
  else if (list.length === 0) list = line.split(" ");
  else arr.push(line.split(" "));
  if (N === arr.length) rl.close();
}).on("close", function () {
  const filterArr = arr.reduce((x, y) => x.concat(y), []);
  console.log(solution(filterArr));
  process.exit();
});
